modeling/datasets_sc.py

```diff
@@ -0,0 +1,369 @@
+import os
+import subprocess
+from tqdm import tqdm
+import random
+from collections import defaultdict
+import pandas as pd
+import numpy as np
+import nibabel as nib
+from scipy.ndimage import zoom
+
+import torch
+from torch.utils.data import Dataset, random_split
+
+from ivadomed.transforms import CenterCrop, RandomAffine, ElasticTransform, \
+     NormalizeInstance, RandomReverse, RandomGamma, RandomBiasField, RandomBlur
+from ivadomed.metrics import dice_score, hausdorff_score, intersection_over_union, \
+    precision_score, recall_score, specificity_score, numeric_score
+from utils import volume2subvolumes, subvolumes2volume
+
```

Comment:
It is important to make sure that the loss is computed the same way for training, validation, and testing. You'll notice that whereas in training and validation we use `DiceLoss` from `ivadomed.losses` as seen in `main.py`, we use `dice_score` from `ivadomed.metrics` for testing. Apparently, there is a difference between the two implementations in ivadomed, namely the addition of `smooth` in `DiceLoss` to handle NaN cases. Therefore, here I would change `dice_score` in the following way:
```suggestion
from ivadomed.losses import DiceLoss
dice_score = DiceLoss(smooth=1.0)
dice_score.__setattr__('__name__', 'dice_score')
```

Reply:
Ah okay, I hadn't noticed that, thanks for pointing it out! Will update